### PR TITLE
Pull in some PTS data for some of the numbers

### PR DESCRIPTION
**Pull request by thelee**
Boy this inflates our project_facts table by an order of magnitude and I'm not altogether clear if this is right.

### CODE REVIEW

File: relational/process_schemaless.py [hunk 1]
```diff
@@ -104,9 +108,10 @@ def store_seen_id(row, header, seen_set):
 ])
 
 
-_FIELDS_TO_CHECK = defaultdict(set)
-_FIELDS_TO_CHECK['ppts'].add('date_opened')
-_FIELDS_TO_CHECK['ppts'].add('date_closed')
+_FIELD_PREDICATE = {}
+_FIELD_PREDICATE[PPTS.NAME] = lambda field: \
+        field == 'date_opened' or field == 'date_closed'
+_FIELD_PREDICATE[PTS.NAME] = lambda field: re.search('date', field)
```
Comment (rajpara11) [inline relational/process_schemaless.py]: Nit: Don't feel strongly but it feels a little safer to just have an explicit of the exact fields we are looking for?

File: relational/test_generators.py [hunk 2]
```diff
@@ -54,29 +58,93 @@ def unit_graph():
...
+    fields = gen_units(proj_missing_proposed_units)
+    # Gets from PPTS because PTS data is incomplete
+    assert _get_value_for_name(fields, 'net_num_units') == '10'
+
+    entries3 = [
+        Entry('1', PPTS.NAME, [NameValue('market_rate_units_net', '10', d)]),
+    ]
+    proj_ppts_only = Project('uuid1', entries3, basic_graph)
+    fields = gen_units(proj_ppts_only)
+    # Gets from PPTS because no ohter choice
```
Comment (rajpara11) [inline relational/test_generators.py]: Nit: Misspelled other

**Comment by thelee** [comment]
did some pretty significant additional changes to fix some bugs, PTAL

File: relational/process_schemaless.py (hunk 1 above)
Comment (thelee) [reply relational/process_schemaless.py]: sounds good - i'll address this in a follow-up commit

File: relational/project.py [hunk 3]
```diff
@@ -86,6 +86,10 @@ class Project:
     """A way to abstract some of the details of handling multiple records for a
     project, from multiple sources."""
 
+    def __str__(self):
+        return '%s => { roots: %s, children %s }' % (self.id, self.roots,
+                                                     self.children)
```
Comment (sbuss) [inline relational/project.py]: A general style nit: args should either all fit on one line or be one-per-line. For example, I'd format this like:

```python
def __str__(self):
    return '%s => { roots: %s, children %s }' % (
        self.id, self.roots, self.children)
```

File: relational/test_generators.py [hunk 4]
```diff
@@ -54,29 +58,116 @@ def unit_graph():
...
+        Entry('1', PPTS.NAME,
+              [NameValue('residential_units_1br_net', '10', d)]),
     ]
     proj_normal = Project('uuid1', entries1, unit_graph)
 
     entries2 = [
-        Entry('1', 'ppts', [NameValue('residential_units_adu_1br_net',
-                                      '1', d)]),
+        Entry('1', PPTS.NAME,
+              [NameValue('residential_units_adu_1br_net', '1', d)]),
```
Comment (sbuss) [inline relational/test_generators.py]: To continue on formatting nits, I'd format this one like:

```python
Entry('1',
      PPTS.NAME,
      [NameValue('residential_units_adu_1br_net', '1', d)]),
```

or

```python
Entry(
    '1',
    PPTS.NAME,
    [NameValue('residential_units_adu_1br_net', '1', d)]),
```

File: relational/project.py (hunk 3 above)
Comment (thelee) [reply relational/project.py]: ooh, thanks, i think i have a lot of  places to fix like this

**Comment by thelee** [comment]
> I had missed your earlier PRs so I had to spend some time reading and working to understand what's going on. In general, there are more tuples, lists, and dicts than I'm comfortable with.

stylistically, named tuples are no good?

>
> I wonder if refactoring it to move these generators onto subclasses of Entry or Project would be cleaner? What have you explored?

nothing - this was the fastest/firstest i spat out :)

> 
> LGTM because my aesthetic concerns aren't important enough to block any work. It could be something we come back to later, if at all.

**Comment by thelee** [comment]
> > LGTM because my aesthetic concerns aren't important enough to block any work. It could be something we come back to later, if at all.

i'm going to go ahead and merge this so we can generate some PTS data (without lots of bugs) but i'll keep an eye on tightening this stuff up.

**Pull request merged**